### ISSUE TITLE
[SPIKE] Investigate github pipelines to replace depreciated deployment logic

### ISSUE THREAD

**Issue by riatzukiza** [raft-tech/TANF-app#1162]
**Description:**

After trying to work on 1059 we discovered that our current deployment flow is being depreciated, and that we can't use it at all for applying terraform updates during deployment flow.

We need to investigate whether or not it will be worth it to automate the application of teraform configs. If its too much work, we will have to document the commands needed to apply terraform changes when they happen.

Timelimit:2-3 hrs

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] _Outcome 1_
- [ ] _Outcome 2_
- [ ] _Outcome 3_
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] _Task 1_
- [ ] _Task 2_
- [ ] _Task 3_
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

- _Note 1_
- _Note 2_
- _Note 3_

**Supporting Documentation:**

https://circleci.com/docs/api/v2/#operation/triggerPipeline
https://github.com/marketplace/actions/circleci-workflow-api-trigger
https://circleci.com/docs/api/v1/#trigger-a-new-job

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_ 

- _Open Question 1_
- _Open Question 2_

**Comment by jtwillis92** [comment]
I was able to confirm that the V2 Pipeline API already works with our project, including running Terraform changes that were previously failing using the old API in #1059. I triggered this using:
```
curl --request POST \
       --url 'https://circleci.com/api/v2/project/gh/raft-tech/TANF-app/pipeline' \
       --header 'Circle-Token: _redacted_' \
       --data '{"branch": "raft-tdp-main"}'
```
The output from the workflows run can be seen here: https://app.circleci.com/pipelines/github/raft-tech/TANF-app/4735/workflows/71b8329b-a4ca-4165-b37f-c30960a8abfb

However, with our current configuration triggering the pipeline this way causes all the defined workflows to run so there are some additional configuration changes we need to make - as outlined in #1260. These changes shouldn't be too much of a lift and will give us the ability to run Terraform with deployments on label assignment.
